syntax/test_helpers.go

```diff
@@ -53,7 +53,7 @@ func AssertCodeEvalsToType(t *testing.T, expected interface{}, code string) bool
 	}
 	codeExpr := pc.CompileExpr(ast)
```

Comment:
I cannot comment at the top of this function.

This test utility functions name doesn't match the comment.
> // AssertCodeEvalsToType asserts that the exprs evaluate to the same value.
Should read
> // AssertCodeEvalsToType asserts that the code evaluates to the expected type.
I checked the call to `AssertExprEvalsToType`, it does indeed only check the type